src/main/com/fulcrologic/rad/routing/html5_history.cljc

```diff
@@ -102,11 +104,11 @@
     (doseq [f listeners]
       (f route (assoc params ::history/direction direction)))))
 
-(defrecord HTML5History [hash-based? listeners generator current-uid prior-route all-events?]
+(defrecord HTML5History [hash-based? listeners generator current-uid prior-route all-events? prefix]
   RouteHistory
   (-push-route! [this route params]
     #?(:cljs
-       (let [url (route->url route params hash-based?)]
+       (let [url (str prefix (route->url route params hash-based?))]
```

Comment:
`route->url` is actually private, and therefore it *is* acceptable to change its signature, since all of the refactorings will happen in this ns only. It's unfortunate that I make url->route public (though ppl wanted it). Both of these really should belong to another browser-based protocol that the `defrecord` implements, so that they can use the config of the history object. 

It is a bit unfortunate that the addition of `prefix` support really does have external API consequence (as did hashing, which was from another contributor). These changes are not localized to the config of the HTML5History object, but instead require peppering the prefix/hash decision all over the app code.

This is kinda crappy.

I think a better solution might be to create a new protocol that HTML5History could implement, so that the route->url and url->route become method of the implementation instead of these externalized things. My original design had them outside like they are because all they did was URL encode/decode of the parameters, and did not rely on the implememtation of the history (just the browser).  With these additions it really is becoming kind of a mess.

I guess the public protocol method `hist/current-route` is really the thing you should use if you're looking for the route in an abstract way, and `url->route` was always meant to be a non-public helper, so I guess that it is a little mess is OK.

So, I guess as long as you clean the user input on `prefix` this is OK.